### DISCUSSION TITLE
431: Request Header Fields Too Large

### DISCUSSION THREAD

**Discussion by codemage8** [vercel/next.js#54531] [Help]
### Summary

After migrating to Next.js 13 from 12,  batched tRPC requests with a very long URL fails with 431 error (The URL length is a total of 7995)

It used to work in Next.js 12 but no longer works in Next.js 13.4.19.

The workaround is here, and setting `maxURLLength` to 6000 worked.
https://trpc.io/docs/client/links/httpBatchLink#setting-a-maximum-url-length

So, it seems like the maximum request header size is changed in Next.js 13.

I would like to know if there's any option to increase it so it works like in Next.js 12.

Thanks.

**Comment by clicktodev** [comment]
looks like it's available in app router https://nextjs.org/docs/app/api-reference/config/next-config-js/reactMaxHeadersLength

mark as solution if this fixes the problem for you

**Comment by sbaechler** [comment]
node limits the maximum header size. It can be increased with the node option `--max-http-eader-size`, e.g. `--max-http-header-size=24000`. But this has to be supported by all servers that sit between the browser and the backend.
